src/DI/ElasticsearchExtension.php

```diff
@@ -5,29 +5,40 @@
 use Elasticsearch\Client;
 use Elasticsearch\ClientBuilder;
 use Nette\DI\CompilerExtension;
-
-class ElasticsearchExtension extends CompilerExtension
+use Nette\DI\Helpers;
+use Nette\Schema\Expect;
+use Nette\Schema\Schema;
+use stdClass;
+
+/**
+ * @property-read stdClass $config
+ */
+final class ElasticsearchExtension extends CompilerExtension
 {
 
-	/** @var mixed[] */
-	private $defaults = [
-		'hosts' => ['localhost'],
-	];
+	public function getConfigSchema(): Schema
+	{
+		return Expect::structure([
+			'hosts' => Expect::arrayOf('string')->required(),
```

Comment:
`$config->hosts === []` does not make sense if value is required

Reply:
Should I remove the `->required()` chain function and leave it with default values or go the other way around that I would drop default values?

Reply:
I don't know elasticsearch. Is `localhost` standard value for host? If it's just for local development, then remove it

Reply:
According to [elasticsearch-php docs (configuration)](https://github.com/elastic/elasticsearch-php/blob/master/docs/configuration.asciidoc) it is not required parameter.

Also at the same time I see an issue that elasticsearch-php supports "extended host configuration" where instead of passing single string passing an associative array with multiple values for easier configuration. And that makes configuration not strict.

I will write schema for the extended configuration

Reply:
Example:
```php
$hosts = [
    // This is effectively equal to: "https://username:password!#$?*abc@foo.com:9200/elastic"
    [
        'host' => 'foo.com',
        'port' => '9200',
        'scheme' => 'https',
        'path' => '/elastic',
        'user' => 'username',
        'pass' => 'password!#$?*abc'
    ],
];
```
is equal to:
```php
$hosts = [
    // This is equal to "http://localhost:9200/"
    [
        'host' => 'localhost',    // Only host is required
    ]
];
```

Reply:
Something like this should work
```php
Expect::arrayOf(Expect::anyOf(Expect::string(), Expect::structure([
  'host' => Expect::string(),
  ...other values
])))
```